File: MTWDM iOS Xamarin/AppChangeTheme/AppChangeTheme/ViewController.cs
using Foundation;
using System;
using UIKit;

namespace AppChangeTheme
{
    public partial class ViewController : UIViewController
    {
        public ViewController(IntPtr handle) : base(handle)
        {
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();
            // Perform any additional setup after loading the view, typically from a nib.

       
            if (TraitCollection.UserInterfaceStyle  == UIUserInterfaceStyle.Light)
            {
                Console.WriteLine("Modo Luz");
            }
            else
            {
                Console.WriteLine("Modo Oscuro");
            }

            this.OverrideUserInterfaceStyle = UIUserInterfaceStyle.Dark;


        }

        partial void onCambiarTema(Foundation.NSObject sender)
        {
            var segmented = sender as UISegmentedControl;


            switch (segmented.SelectedSegment)
            {
                case 0:
                    this.OverrideUserInterfaceStyle = UIUserInterfaceStyle.Dark;
                    break;
                case 1:
                    this.OverrideUserInterfaceStyle = UIUserInterfaceStyle.Light;
                    break;
                default:
                    break;
            }

        }

        public override void DidReceiveMemoryWarning()
        {
            base.DidReceiveMemoryWarning();
            // Release any cached data, images, etc that aren't in use.
        }
    }
}

File: MTWDM iOS Swift/appChangeTheme/appChangeTheme/ViewController.swift
//
//  ViewController.swift
//  appChangeTheme
//
//  Created by Felipe Hernandez on 9/25/19.
//  Copyright © 2019 Felipe Hernandez. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    
        
        if traitCollection.userInterfaceStyle == .light {
                  print("Modo Luz")
              } else {
                  print("Modo Oscuro")
              }
        
     self.overrideUserInterfaceStyle = .dark
    
        
    }
    
   /* override var preferredStatusBarStyle: UIStatusBarStyle {
        return  .darkContent  //return ..lightContent
    }*/

  
    @IBAction func onCambiarTema(_ sender: UISegmentedControl) {
        
        switch sender.selectedSegmentIndex {
        case 0:
             self.overrideUserInterfaceStyle = .dark
        case 1:
            self.overrideUserInterfaceStyle = .light
        default:
             break
        }
    }

}

